cmd/metacache-bucket.go

```diff
@@ -193,23 +192,18 @@ func (b *bucketMetacache) findCache(o listPathOptions) metacache {
 		debugPrint = logger.Info
 	}
 
-	// Grab a write lock, since we create one if we cannot find one.
-	if o.Create {
-		b.mu.Lock()
-		defer b.mu.Unlock()
-	} else {
-		b.mu.RLock()
-		defer b.mu.RUnlock()
-	}
+	// Hold read locks in the read phase.
```

Comment:
No. This is meant to serialize requests.

Consider 50 requests coming in for the same path. If they are not serialized each of them will create a new cache and crawl the same data.

This loop is meant to be fast enough to keep up, and even if there is a few milliseconds of wait the rest of the system will have more than enough to do.